src/algorithms/pathfinders/dijkstra.ts

```diff
@@ -0,0 +1,111 @@
+import EdgeCost from '../models/edgecost';
+import Graph from '../models/graph';
+import Vertex from '../models/vertex';
+/**
+ * Implements Dijkstra's algorithm to find the shortest path between 2 vertices
+ */
+export default class DijkstraPathfinder {
+    /**
+     * Stores best known distances from starting vertex to all the others
+     */
+    distances: Map<number, {
+        distance: number;
+        sourceID: number;
+    }>;
+
+    /**
+     * stores all unvisited vertices
+     */
+    unvisited: Set<number>;
+
+    /**
+     * Returns an array of vertex IDs that represents the shortest path between 2 vertices
+     * @param startID : ID of starting vertex
+     * @param endID : ID of destination vertex
+     * @param mygraph : relevant graph
+     */
+    FindPath(startID: number, endID: number, mygraph: Graph): number[] {
+        this.Initialize(mygraph);
+        this.distances.set(startID, {distance: 0, sourceID: -9001});
+        this.unvisited.delete(startID);
+        let currentVertex = mygraph.getVertex(startID);
+        while (this.unvisited.size > 0) {
+            this.updateDistances(currentVertex);
+            currentVertex = this.getNextVertex(mygraph);
+            if (currentVertex.id === endID) {
+                break;
+            }
+        }
+        return this.producePath(startID, endID);
+    }
+
+    /**
+     * Initializes data structures for finding paths in the given graph
+     * @param mygraph : relevant graph
+     */
+    Initialize(mygraph: Graph): void {
+        this.unvisited = mygraph.getVertexIDs();
+        this.distances = new  Map<number, {
+            distance: number;
+            sourceID: number;
+        }>();
+        for (const vertexID of this.unvisited) {
+            this.distances.set(vertexID, {distance: global.Infinity, sourceID: -9001});
+        }
+    }
+
+    /**
+     * Finds the next best vertex of the graph to visit
+     * @param mygraph : relevant graph
+     */
+    getNextVertex(mygraph: Graph): Vertex {

```

Comment:
This is a part that would be unnecessary if you used priority queue and instead of looping through the whole set, just pop the first element in an array.